webpack/assets/javascripts/react_app/common/urlHelpers.js

```diff
@@ -2,6 +2,21 @@ import URI from 'urijs';
 
 import { visit } from '../../foreman_navigation';
 
+/**
+ * Build a url with query params
+ * @param {String} url - the base url i.e `/hosts`
+ * @param {Object} searchQuery - the query params, i.e {'per_page': 4, 'page': 2}
+ */
+export const urlWithQueryParams = (url, queryParams) => {
+  const searchParams = new URLSearchParams('');
+
+  queryParams &&
+    Object.entries(queryParams).forEach(([param, value]) =>
+      searchParams.append(param, value)
+    );
+  return `${url}?${searchParams.toString()}`;
+};
+
```

Comment:
just because `URLSearchParams` is native, don't mind to use URI.js but i think we can stop using it with modern JS (like lodash)

Reply:
When I see us building URLs manually like this function, i tend to think that we are doing something wrong and are most likely missing some edge cases. Just because something can be done in native JS, doesn't necessarily mean we have to manually implement it if there is a good library that is well tested and abstracts away the complexity.
If we use URI.js, this whole function turns into:
```js
URI(url).search(queryParams)
```
which is much more readable and doesn't even necessarily warrant creating a helper function for it.